library/ssl_client.c

```diff
@@ -641,6 +643,18 @@ static int ssl_write_client_hello_body( mbedtls_ssl_context *ssl,
     }
 #endif /* MBEDTLS_SSL_PROTO_TLS1_2 */
 
+#if defined(MBEDTLS_SSL_PROTO_TLS1_3) && \
+    defined(MBEDTLS_KEY_EXCHANGE_SOME_PSK_ENABLED)
+    if( propose_tls13 && mbedtls_ssl_conf_tls13_some_psk_enabled( ssl ) )
+    {
```

Comment:
```suggestion
    /* Writing offered_psks ( ClientHello version of pre_shared_key ) MUST be
     * the LAST extension. Otherwise, `transcript data` will be corrupted 
     */
    if( propose_tls13 && mbedtls_ssl_conf_tls13_some_psk_enabled( ssl ) )
    {
```